### ISSUE TITLE
addressDiscovery: "event" dictionary removal from AS3 declaration JSON errors out

### ISSUE THREAD

**Issue by manukanade** [F5Networks/f5-appsvcs-extension#198]
Do you already have an issue opened with F5 support?
Github Issues are consistently monitored by F5 staff, but should be considered as best effort only and you should not expect to receive the same level of response as provided by F5 Support. Please open an case with F5 if this is a critical issue.

ISSUE TYPE
Bug Report

AS3 BUILD/ VERSION
AS3 v 3.15.0-6

Build Details :
BIG-IP v 13.1.1.3 ( Physical , Standalone)

SUMMARY
Dynamic discovery removal from AS3 JSON results in error when nodes exist in dynamic discovery endpoint

STEPS TO REPRODUCE
1. POST https://BIG-IP/mgmt/shared/appsvcs/declare -  Posted an AS3 declaration JSON with service-discovery entry added in it: {"addressDiscovery": "event", "servicePort": 8080} 
{
    "schemaVersion": "3.15.0",
    "class": "ADC",
    "partition1": {
        "application1": {
            "web_pool": {
                "class": "Pool",
                "members": [
                    {
                        "serverAddresses": [],
                        "servicePort": 80
                    },
                    {
                        "addressDiscovery": "event",
                        "servicePort": 8080
                    }
                ],
                "monitors": [
                    "http"
                ]
            },
            "class": "Application",
            "template": "http",
            "serviceMain": {
                "enable": false,
                "class": "Service_HTTP",
                "pool": "web_pool",
                "virtualAddresses": [
                    "10.10.10.10"
                ]
            }
        },
        "class": "Tenant"
    },
    "id": "0dc4041e-7fb6-475c-bc65-3961cdd0b0d8"
}

2. Checked that service discovery endpoint got created
GET https://BIG-IP/mgmt/shared/service-discovery/task/~partition1~application1~web_pool
{"code":200,"result":{"id":"\~partition1\~application1\~web_pool","updateInterval":0,"provider":"event","providerOptions":{"type":"event","nodeList":[]},"resources":[{"type":"pool","path":"/partition1/application1/web_pool","options":{"servicePort":8080,"connectionLimit":0,"rateLimit":"disabled","dynamicRatio":1,"ratio":1,"priorityGroup":0,"monitor":"default"},"requiresNodes":true}],"kind":"shared:service-discovery:taskstate","lastRun":"2019-11-22T06:08:10.864Z","nodePrefix":"/partition1/"}}

3. Added a list of nodes to service discovery endpoint: POST to https://BIG-IP/mgmt/shared/service-discovery/task/~partition1~application1~web_pool/nodes
Body: [{"id": "192.168.10.10", "ip": "192.168.10.10"}, {"id": "192.168.10.11", "ip": "192.168.10.11"}]
-> Got 200 OK

4. POST https://BIG-IP/mgmt/shared/appsvcs/declare: **Removed Service Discovery dictionary from AS3 declaration JSON**
{
    "schemaVersion": "3.15.0",
    "class": "ADC",
    "partition1": {
        "application1": {
            "web_pool": {
                "class": "Pool",
                "members": [
                    {
                        "serverAddresses": [],
                        "servicePort": 80
                    }                    
                ],
                "monitors": [
                    "http"
                ]
            },
            "class": "Application",
            "template": "http",
            "serviceMain": {
                "enable": false,
                "class": "Service_HTTP",
                "pool": "web_pool",
                "virtualAddresses": [
                    "10.10.10.10"
                ]
            }
        },
        "class": "Tenant"
    },
    "id": "0dc4041e-7fb6-475c-bc65-3961cdd0b0d9"
}

5. After step 4, was expecting a success response and service discovery entry to be removed - but got an error:
{
    "results": [
        {
            "code": 422,
            "message": "declaration failed",
            "response": "01020036:3: The requested Pool Member (/partition1/application1/web_pool /partition1/192.168.10.10 8080) was not found.",
            "host": "localhost",
            "tenant": "partition1",
            "runTime": 8007
        }
    ],
    "declaration": {
        "partition1": {
            "application1": {
                "web_pool": {
                    "class": "Pool",
                    "members": [
                        {
                            "serverAddresses": [],
                            "servicePort": 80
                        },
                        {
                            "addressDiscovery": "event",
                            "servicePort": 8080
                        }
                    ],
                    "monitors": [
                        "http"
                    ]
                },
                "class": "Application",
                "template": "http",
                "serviceMain": {
                    "enable": false,
                    "class": "Service_HTTP",
                    "pool": "web_pool",
                    "virtualAddresses": [
                        "10.10.10.10"
                    ]
                }
            },
            "class": "Tenant"
        },
        "schemaVersion": "3.15.0",
        "class": "ADC",
        "id": "0dc4041e-7fb6-475c-bc65-3961cdd0b0d9",
        "updateMode": "selective",
        "controls": {
            "archiveTimestamp": "2019-11-22T06:18:09.995Z"
        }
    },
    "code": 422
}

When I checked the service discovery endpoint https://BIG-IP/mgmt/shared/service-discovery/task/~partition1~application1~web_pool, it was removed as expected:
{"code":404,"message":"status:404, body:{\"code\":404,\"error\":\"\~partition1\~application1\~web_pool was not found\"}","originalRequestBody":"{\"code\":404,\"error\":\"\~partition1\~application1\~web_pool was not found\"}","referer":"10.107.0.5","restOperationId":5999053,"kind":":resterrorresponse"}

**But I was not expecting the error in AS3 JSON post above which was: 
            "code": 422,
            "message": "declaration failed",
            "response": "01020036:3: The requested Pool Member (/partition1/application1/web_pool /partition1/192.168.10.10 8080) was not found.",**
			
			
6. After re-posting the same JSON with removed service discovery as in step 5, it was finally successful and got 200 OK

**Comment by manukanade** [comment]
This is a duplicate of https://github.com/F5Networks/f5-appsvcs-extension/issues/185. Closing
